src/base/types.h

```diff
@@ -113,12 +113,20 @@ typedef unordered_map<JobID_t, JobDescriptor,
         boost::hash<boost::uuids::uuid> > JobMap_t;
 typedef unordered_map<JobID_t, TaskGraph*,
         boost::hash<boost::uuids::uuid> > TaskGraphMap_t;*/
+typedef thread_safe::set<ResourceID_t> ResourceSet_t;
+typedef thread_safe::map<ResourceID_t, uint64_t> ResourceFrequencyMap_t;
+typedef thread_safe::map<ResourceID_t, EquivClass_t> ResourceEquivClassMap_t;
+typedef thread_safe::map<ResourceID_t, int64_t> ResourceCostMap_t;
 typedef thread_safe::map<ResourceID_t, ResourceStatus*> ResourceMap_t;
 typedef thread_safe::map<JobID_t, JobDescriptor> JobMap_t;
 typedef thread_safe::map<JobID_t, TaskGraph*> TaskGraphMap_t;
 #else
 typedef uint64_t ResourceID_t;
 typedef uint64_t JobID_t;
+typedef unordered_set<ResourceID_t> ResourceSet_t;
+typedef unordered_map<ResourceID_t, uint64_t> ResourceFrequencyMap_t;
+typedef unordered_map<ResourceID_t, EquivClass_t> ResourceEquivClassMap_t;
+typedef unordered_map<ResourceID_t, int64_t> ResourceCostMap_t;
```

Comment:
It turns out that the simulated DFS class also uses this type. For now, we have two typedefs in different places for this, but that's clearly not ideal. We'll need to think about a shared header in the future. However the places that use this are in different modules ("sim" and "cost_models"), so there is no trivial solution.